web3/middleware/exception_retry_request.py

```diff
@@ -87,11 +89,16 @@
 ]
 
 
-def check_if_retry_on_failure(method: RPCEndpoint) -> bool:
+def check_if_retry_on_failure(
+    method: str, allow_list: Optional[List[str]] = None
+) -> bool:
+    if allow_list is None:
```

Comment:
What do you think about using the default if `allow_list` has zero items as well as `None`?

Reply:
I think I could go either way on that but it seems a bit disingenuous. I don't think it's ideal to submit an empty list for a method like this but also in python I'd always expect if I pass in a kwarg, that is what's used / overrides the default kwarg value. What are your thoughts there? 👀

Reply:
Logically I see your point, if looking for an item in an empty list, I would expect it to return false. I'm good leaving it as is!